apisix/cli/ngx_tpl.lua

```diff
@@ -596,7 +596,7 @@ http {
             proxy_cache                         $upstream_cache_zone;
             proxy_cache_valid                   any {% if proxy_cache.cache_ttl then %} {* proxy_cache.cache_ttl *} {% else %} 10s {% end %};
             proxy_cache_min_uses                1;
-            proxy_cache_methods                 GET HEAD;
+            proxy_cache_methods                 GET HEAD POST;
```

Comment:
Nginx by default doesn't enable the cache for POST. Why change this behavior here?

Reply:
It doesn't affect the default behavior of APISIX, which means that user can specify the `POST` method on APISIX.

Reply:
Is it necessary to cache a POST request?

The only I can think of for caching POST requests is that some internal specification forces the consistent use of POST requests……

Reply:
> It doesn't affect the default behavior of APISIX, which means that user can specify the `POST` method on APISIX.

Here you change the Nginx template, without a configurable setting. This is a change for default havior.

Reply:
> @agile6v
> Could you sync it to APISIX.pm?

done

Reply:
> Is it necessary to cache a POST request?
> 
> The only I can think of for caching POST requests is that some internal specification forces the consistent use of POST requests……

Yes. This is a very rare scenario. The users don't enable the method on APISIX,  it doesn't take effect.  If users enable it, they need to know what they are doing.

Reply:
> > It doesn't affect the default behavior of APISIX, which means that user can specify the `POST` method on APISIX.
> 
> Here you change the Nginx template, without a configurable setting. This is a change for default havior.

Whether to enable is [here](https://github.com/agile6v/apisix/blob/master/docs/en/latest/plugins/proxy-cache.md) (cache_method)